apps/html-renderer/packages/generic/src/renderHTML.ts

```diff
@@ -0,0 +1,497 @@
+import { BrowserWindow, app, ipcMain } from 'electron'
+import { spawn } from 'child_process'
+import * as path from 'path'
+import * as fs from 'fs'
+import { LoggerInstance, escapeFilePath, getFFMpegExecutable, testFFMpeg } from '@sofie-package-manager/api'
+import { sleep } from '@sofie-automation/shared-lib/dist/lib/lib'
+
+export interface RenderHTMLOptions {
+	logger: LoggerInstance
+	/** URL to the web page to render */
+	url: string
+	/** Width of the window */
+	width?: number
+	/** Height of the window */
+	height?: number
+	/** Zoom factor */
+	zoom?: number
+	/** Background color, default to black */
+	backgroundColor?: string
+
+	tempFolder?: string
+	outputFolder?: string
+	/** Scripts to execute */
+	scripts: {
+		fcn?: (options: { webContents: Electron.WebContents }) => void | Promise<void>
+		executeJs?: string
+		logInfo?: string
+		takeScreenshot?: {
+			/** PNG file */
+			name: string
+		}
+		startRecording?: {
+			name: string
+			full?: boolean
+			cropped?: boolean
+		}
+		wait: number
+	}[]
+	userAgent?: string
+	/** Interactive mode, overrides scripts */
+	interactive?: (api: InteractiveAPI) => Promise<void>
+}
+export async function renderHTML(options: RenderHTMLOptions): Promise<{
+	app: Electron.App
+	exitCode: number
+}> {
```

Comment:
Could this function be broken up a little bit? It's very long, has loads of functions-within-functions, maybe this could be straightened up a bit?